### DISCUSSION TITLE
How-To start GUI app in macOS Monterey?

### DISCUSSION THREAD

**Discussion by GETandSELECT** [nicotine-plus/nicotine-plus#1675] [Q&A] [answered]
Hi

I did a 

`brew install nicotine-plus` and lots of dependencies got installed. 

How-To start GUI app in macOS Monterey?

**Comment by slook** [comment]
There are problems with N+ on latest MacOS at the moment see https://github.com/nicotine-plus/nicotine-plus/issues/1660 work is in progress to remedy this.

What occurs when you attempt to run the application?

**Comment by GETandSELECT** [comment]
I don't even know how-to start the GUI 

```
./nicotine
[2021-11-06 07:43:33] You need to specify a username and password before connecting...
[2021-11-06 07:43:33] UPnP exception: [Errno 35] Resource temporarily unavailable
[2021-11-06 07:43:33] Failed to automate the creation of UPnP Port Mapping rule.
[2021-11-06 07:43:33] Listening on port 2234
^C%
```

**Reply by slook** [reply]
The GUI should launch by itself and show the Setup Assistant which allows you to choose a username etc.

You are experiencing the bug in https://github.com/nicotine-plus/nicotine-plus/issues/1660 which seems to cause the GUI to fail. You could help us to fix this see below.

**Reply by slook** [reply]
Make sure you are not somehow running `nicotine` with the `-s` switch.

**Comment by francisuk1989** [comment]
Someone eles is having issues with MacOS Monteray https://github.com/nicotine-plus/nicotine-plus/issues/1660 but this sounds like a GTK 3 issue however MacOS Monteray is supported by Brew and providing you **dont** have the Apple M1 chipset, See @mathiascode note here https://github.com/nicotine-plus/nicotine-plus/issues/1475#issuecomment-944769368 then you could always give homebrew a try with GTK4

So in that case, if you willing to install Brew/homebrew then i would go by as i know the following are supported what uses nicotine+

So we know that
1) python@3.9 is supported by Monteray https://formulae.brew.sh/formula/python@3.9
2) py3cairo is supported by Monteray  https://formulae.brew.sh/formula/py3cairo
3) pygobject3 supported by Monteray  https://formulae.brew.sh/formula/pygobject3
4) gtk4 supported by Monteray  https://formulae.brew.sh/formula/gtk4

-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

1) Install Brew
`https://brew.sh`

2) install the following
```
brew install python@3.9
brew install py3cairo
brew install pygobject3
brew install gtk4
brew install adwaita-icon-theme

```
3)
```
git clone https://github.com/nicotine-plus/nicotine-plus
cd nicotine-plus
./NICOTINE_GTK_VERSION=4 nicotine
```

let me know if it works.

**Reply by slook** [reply]
What is the meaning of `"-s"` on Line 32 in the Homebrew Formulae?

https://github.com/Homebrew/homebrew-core/blob/c4afdfbc730ffc04cde7588b42adbbde105f3f00/Formula/nicotine-plus.rb#L32

`      exec bin/"nicotine", "-s"`

Is that because it is for testing or some other purpose for the syntax of exec? I don't get it.

  _-s, --hidden          start the program without showing window_

Running nicotine with the -s switch would look like the OP's console output above https://github.com/nicotine-plus/nicotine-plus/discussions/1675#discussioncomment-1598376

**Reply by francisuk1989** [reply]
Interesting,,, as i though --headless would of start the program without showing window

**Reply by slook** [reply]
`-s` is the shorthand for `--headless` (s for silent I suppose, since -h is --help)

`exec bin/"nicotine", "-s"` is in the Formula for package testing purposes. I don't think it is related the OP's problem.

**Reply by slook** [reply]
```
$ ./nicotine --help
usage: nicotine [-h] [-c file] [-u dir] [-p dir] [-t] [-d] [-s] [-b ip]
                [-l port] [-r] [-n] [-v]

Nicotine+ is a Soulseek client

optional arguments:
  -h, --help            show this help message and exit
 ...
  -s, --hidden          start the program without showing window
 ...
  -n, --headless        start the program in headless mode (no GUI)
...
```

**Reply by slook** [reply]
Interesting stuff here https://github.com/pyinstaller/pyinstaller/issues/6354 might be useful to Mac and brew users, there's a conflict with a Gtk3 library that can be worked around.

**Comment by francisuk1989** [comment]
Sorry to tag you @afincato - I don't suppose you have Mac OS "Monteray" and tired this before with GTK 4?
https://github.com/nicotine-plus/nicotine-plus/discussions/1675#discussioncomment-1598381

**Reply by sonn-gamm** [reply]
No problem @francisuk1989. I haven't upgrade to Monterey yet and cannot help you with testing / debugging.

**Reply by francisuk1989** [reply]
Not a problem @afincato thanks anyway :+1:

**Comment by slook** [comment]
@GETandSELECT I think the [latest Nicotine+](https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#macos-catalina1015-and-newer) 3.2.0.rc1 may have recently overcome a possibly related issue that might help with the problem you were having. Are you able to confirm if it works now or if this still needs something else?

**Comment by Inso-m-niaC** [comment, marked as answer]
this issue has been patched in homebrew, run command `brew update`

**Reply by francisuk1989** [reply]
```
brew update
brew upgrade
brew cleanup
```

**Reply by GETandSELECT** [reply]
thanks, working for me
